04-mpu6050/script/mpu6050.sh

```diff
@@ -1,24 +1,24 @@
 #!/bin/bash
 
-started=`i2cget -y 1 0x68 0x6b`
-started="${started: -2}"
-
-if [ $started -ne 0 ]
-then 
-	i2cset -y 1 0x68 0x6b 0
-fi
+i2cset -y 1 0x68 0x6b 0
 
 while [ true ]
 do
 	tempHexH=`i2cget -y 1 0x68 0x41`
 	tempHexL=`i2cget -y 1 0x68 0x42`
-
-	temperature=$(( -(((tempHexH << 8) + tempHexL) ^ ((1 << 0x10) - 1)) ))
+	
+	sign=$(( (tempHexH & 0x80) >> 7 ))
+	temperature=$(( (tempHexH << 8) + tempHexL ))
+	
+	if [ $sign = 1 ]
+	then
+		temperature=$(( -($temperature ^ 0xffff) ))
```

Comment:
This conversation is not correct. The implemented function do conversation to 1's complement representation.
For 2's complement You can use
-(0x10000 - $temperature)
it is the same like
$temperature - 0x10000
instead of -($temperature  0xffff)